src/k8s/pkg/k8sd/api/worker_handler.go

```diff
@@ -0,0 +1,26 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/canonical/k8s/pkg/snap"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/state"
+)
+
+func RestrictWorkers(s *state.State, r *http.Request) response.Response {
+	snap := snap.SnapFromContext(s.Context)
+
+	isWorker, err := snaputil.IsWorker(snap)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("failed to check if node is a worker: %w", err))
+	}
+
+	if isWorker {
+		return response.InternalError(fmt.Errorf("this action is restricted on workers"))
```

Comment:
We should return a 403 Forbidden here instead. That way, we can give nice error messages in the CLI.